kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/EncryptionScheme.java

```diff
@@ -6,23 +6,31 @@
 
 package io.kroxylicious.filter.encryption;
 
+import java.util.EnumSet;
 import java.util.Objects;
 import java.util.Set;
 
+import edu.umd.cs.findbugs.annotations.NonNull;
+
 /**
  * Describes how a record should be encrypted
  * @param kekId The KEK identifier to be used. Not null.
- * @param recordFields The fields of the record that should be encrypted with the given KEK. Neither null nor empty.
+ * @param recordFields The fields of the record that should be encrypted with the given KEK. The empty set indicates that encryption should not be performed.
  * @param <K> The type of KEK identifier.
  */
 public record EncryptionScheme<K>(
-                                  K kekId,
-                                  Set<RecordField> recordFields) {
+                                  @NonNull K kekId,
+                                  @NonNull Set<RecordField> recordFields) {
     public EncryptionScheme {
         Objects.requireNonNull(kekId);
-        if (Objects.requireNonNull(recordFields).isEmpty()) {
-            throw new IllegalArgumentException();
-        }
+        Objects.requireNonNull(recordFields);
+    }
+
+    public static <K> EncryptionScheme<K> unencryptedScheme(K kekId) {
```

Comment:
The existence of this factory implies we should also have 
`RecordScheme(K kekId)` and `RecordAndHeadersScheme(K kekId)` 

Which feels strange to say the least. 

So maybe instead of/as well as giving meaning to the empty set we should expand `RecordField` it include `None`? 

cc: @tombentley

Reply:
>How would you disallow meaningless combinations Set.of(NONE,RECORD_VALUE)?

On the one hand I ask why do we need to disallow it? Just because it doesn't have obvious value to us doesn't mean it should be banned. Maybe there is a some future usage where that could be useful. Also looking at our current usage there is no reason to ban the combination as we pick specific types to operate on. 

On the other hand your right as an API it does invite people to setup strange and potentially contradictory instructions.